src/Recommendations/RuleModel.php

```diff
@@ -74,4 +75,24 @@ public function getPublished()
     {
         return $this->published;
     }
+
+    public function __toString()
+    {
+        return json_encode($this);
+    }
+
+    public function equalTo(RuleModel $model)
+    {
+        return $model->getId() === $this->getId() && $model->getType() === $this->getType();
+    }
+
+    public function jsonSerialize()
+    {
+        return [
+            'id' => $this->id,
+            'type' => $this->type,
+            'published' => $this->published,
+            'isSynthetic' => $this->isSynthetic,
+        ];
+    }
```

Comment:
Just to clarify the intentions of this, when we debug to the console with an object it tries to grab it as a string, I figured the best representation will be just plain JSON. Just overriding the `__toString()` got messy with nested objects.